python/hackathon_app/sheets.py

```diff
@@ -21,121 +25,170 @@ def __init__(self, spreadsheet_id: str, cred_file: str):
         )
 
         service = discovery.build("sheets", "v4", credentials=credentials)
-        self.spreadsheet = service.spreadsheets().values()
+        client = service.spreadsheets().values()
         self.id = spreadsheet_id
+        self.hackathons = Hackathons(client, spreadsheet_id)
+        self.registrations = Registrations(client, spreadsheet_id)
+        self.users = Users(client, spreadsheet_id)
 
-    def get_hackathons(self):
+    def get_hackathons(self) -> Sequence["Hackathon"]:
         """Get names of active hackathons."""
-        hackathons_table = self.spreadsheet.get(
-            spreadsheetId=self.id, range="hackathons!A1:end"
-        ).execute()
-        data = hackathons_table["values"][1:]
-        date_index = hackathons_table["values"][0].index("date")
-        hackathons = []
-        for row in data:
-            if datetime.strptime(row[date_index], "%m/%d/%Y") >= datetime.now():
-                hackathons.append(row[1])
-        return hackathons
-
-    def register_user(
-        self,
-        hackathon: str,
-        first_name: str,
-        last_name: str,
-        email: str,
-        company: str,
-        country: str,
-        tshirt_size: str,
-    ):
+        hackathons = self.hackathons.rows()
+        result = []
+        for hackathon in hackathons:
+            if hackathon.date >= datetime.datetime.now():
+                result.append(hackathon)
+        return result
+
+    def register_user(self, hackathon: str, user: "User"):
         """Register user to a hackathon"""
-        if not self._is_created(email):
-            self.create_user(
-                first_name, last_name, email, company, country, tshirt_size
-            )
-        if not self._is_registered(email, hackathon):
-            self._register_user(email, hackathon)
-
-    def _is_created(self, email: str) -> bool:
-        """Check if user already exists in the users sheet."""
-        users = self.get_users()
+        if not self.users.is_created(user):
+            self.users.create(user)
+        registrant = Registrant(
+            user_email=user.email,
+            hackathon_name=hackathon,
+            date_registered=datetime.datetime.now(),
+            attended=None,
+        )
+        if not self.registrations.is_registered(registrant):
+            self.registrations.register(registrant)
+
+
+class WhollySheet:
+    def __init__(self, client, spreadsheet_id, sheet_name):
+        self.client = client
+        self.spreadsheet_id = spreadsheet_id
+        self.range = sheet_name + "!A1:end"
```

Comment:
```suggestion
        self.range = f"{sheet_name}!A1:end"
```
In languages that have an alternative I prefer to keep my arithmetic operators to numbers :-)